control/rlocus.py

```diff
@@ -157,29 +290,33 @@ def _systopoly1d(sys):
     # Check to see if num, den are already polynomials; otherwise convert
     if (not isinstance(nump, poly1d)):
         nump = poly1d(nump)
+
     if (not isinstance(denp, poly1d)):
         denp = poly1d(denp)
 
     return (nump, denp)
 
 
-def _RLFindRoots(sys, kvect):
+def _RLFindRoots(nump, denp, kvect):
     """Find the roots for the root locus."""
 
     # Convert numerator and denominator to polynomials if they aren't
-    (nump, denp) = _systopoly1d(sys)
-
     roots = []
     for k in kvect:
         curpoly = denp + k * nump
         curroots = curpoly.r
+        if len(curroots) < denp.order:
+            curroots = np.insert(curroots, len(curroots), np.inf)  # if i have less poles than open loop is because  i
```

Comment:
The comment here overflows into the next line. For ease of reading, it should be moved to the line before. This style for block comments is given [in PEP 8](https://www.python.org/dev/peps/pep-0008/#block-comments). E.g., after I edit the comment text a little,
```python
if len(curroots) < denp.order:
    # If I have fewer poles than open loop, it is because I
    # have one at infinity.
    curroots = np.insert(curroots, len(curroots), np.inf)
```